src/PKSim.Core/Snapshots/Services/SnapshotSerializer.cs

```diff
@@ -24,9 +28,21 @@ public void Serialize(object snapshot, string fileName)
          File.WriteAllText(fileName, JsonConvert.SerializeObject(snapshot, Formatting.Indented, _settings));
       }
 
-      public T Deserialize<T>(string fileName)
+      public object[] DeserializeAsArray(string fileName, Type snapshotType)
       {
-         return JsonConvert.DeserializeObject<T>(File.ReadAllText(fileName), _settings);
+         var deserializedSnapshot = JsonConvert.DeserializeObject(File.ReadAllText(fileName), _settings);
+
+         //TODO. We should validate that the json actually matches our type. If not, we could throw an explicit error
```

Comment:
this is an internal Json object. In theory I could have selected a file with compound information while trying to load a formulation. This is where we will use json schema to ensure that the json structure matches our object. If not, we will throw an exception with hopefully some useful info